internal/storage/v2/elasticsearch/tracestore/from_dbmodel.go

```diff
@@ -183,37 +93,71 @@ type scope struct {
 	name, version string
 }
 
-func jSpansToInternal(spans []*model.Span, dest ptrace.ScopeSpansSlice) {
-	spansByLibrary := make(map[scope]ptrace.SpanSlice)
-
+func jSpansToInternal(spans []*dbmodel.Span, resourceSpans ptrace.ResourceSpansSlice) error {
 	for _, span := range spans {
 		if span == nil || reflect.DeepEqual(span, blankJaegerProtoSpan) {
 			continue
 		}
+		resourceSpan := resourceSpans.AppendEmpty()
+		dest := resourceSpan.ScopeSpans()
+		jProcessToInternalResource(span.Process, resourceSpan.Resource())
 		il := getScope(span)
-		sps, found := spansByLibrary[il]
-		if !found {
-			ss := dest.AppendEmpty()
-			ss.Scope().SetName(il.name)
-			ss.Scope().SetVersion(il.version)
-			sps = ss.Spans()
-			spansByLibrary[il] = sps
+		ss := dest.AppendEmpty()
+		ss.Scope().SetName(il.name)
+		ss.Scope().SetVersion(il.version)
+		sps := ss.Spans()
+		err := jSpanToInternal(span, sps.AppendEmpty())
+		if err != nil {
+			return err
 		}
-		jSpanToInternal(span, sps.AppendEmpty())
 	}
+	return nil
 }
 
-func jSpanToInternal(span *model.Span, dest ptrace.Span) {
-	dest.SetTraceID(idutils.UInt64ToTraceID(span.TraceID.High, span.TraceID.Low))
-	dest.SetSpanID(idutils.UInt64ToSpanID(uint64(span.SpanID)))
-	dest.SetName(span.OperationName)
-	dest.SetStartTimestamp(pcommon.NewTimestampFromTime(span.StartTime))
-	dest.SetEndTimestamp(pcommon.NewTimestampFromTime(span.StartTime.Add(span.Duration)))
+func getTraceIdFromDbTraceId(dbTraceId dbmodel.TraceID) (pcommon.TraceID, error) {
```

Comment:
try not to mix new functions with old - put new functions somewhere at the end so that they don't mess up the diff